File: Client/orgTable/code/ProtocolVip.lua
local descriptor = require "descriptor"

module "ProtocolVip"

Vip = descriptor.def_struct("Vip", 
	{
		descriptor.def_scalar_field("level", descriptor.type_uint8, 0),
		descriptor.def_scalar_field("exp", descriptor.type_uint32, 0),
	}
)

SceneVipBuyItemReq = descriptor.def_message("SceneVipBuyItemReq", 503301, 
	{
		descriptor.def_scalar_field("vipLevel", descriptor.type_uint8, 0),
	}
)

SceneVipBuyItemRes = descriptor.def_message("SceneVipBuyItemRes", 503302, 
	{
		descriptor.def_scalar_field("result", descriptor.type_uint32, 0),
	}
)

File: Client/orgTable/code/ProtocolVip.cs
using System;
using System.Text;

namespace Protocol
{
	public class Vip : Protocol.IProtocolStream
	{
		public byte level;
		public UInt32 exp;

		#region METHOD

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_int8(buffer, ref pos_, level);
				BaseDLL.encode_uint32(buffer, ref pos_, exp);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_int8(buffer, ref pos_, ref level);
				BaseDLL.decode_uint32(buffer, ref pos_, ref exp);
			}
		#endregion

	}

	[Protocol]
	public class SceneVipBuyItemReq : Protocol.IProtocolStream, Protocol.IGetMsgID
	{
		public const UInt32 MsgID = 503301;
		public byte vipLevel;

		#region METHOD
			public UInt32 GetMsgID()
			{
				return MsgID;
			}

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_int8(buffer, ref pos_, vipLevel);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_int8(buffer, ref pos_, ref vipLevel);
			}
		#endregion

	}

	[Protocol]
	public class SceneVipBuyItemRes : Protocol.IProtocolStream, Protocol.IGetMsgID
	{
		public const UInt32 MsgID = 503302;
		public UInt32 result;

		#region METHOD
			public UInt32 GetMsgID()
			{
				return MsgID;
			}

			public void encode(byte[] buffer, ref int pos_)
			{
				BaseDLL.encode_uint32(buffer, ref pos_, result);
			}

			public void decode(byte[] buffer, ref int pos_)
			{
				BaseDLL.decode_uint32(buffer, ref pos_, ref result);
			}
		#endregion

	}

}
